src/code/z_player_lib.c

```diff
@@ -1,153 +1,1046 @@
+    Player* player = GET_PLAYER(globalCtx);
+
+    return (player->transformation == PLAYER_FORM_HUMAN) && (player->currentShield == PLAYER_SHIELD_MIRROR_SHIELD);
+}
+
+s32 Player_HasMirrorShieldSetToDraw(GlobalContext* globalCtx) {
+    Player* player = GET_PLAYER(globalCtx);
+
+    return (player->transformation == PLAYER_FORM_HUMAN) && (player->rightHandType == 8) &&
+           (player->currentShield == PLAYER_SHIELD_MIRROR_SHIELD);
+}
+
+s32 Player_HoldsHookshot(Player* player) {
+    return player->itemActionParam == PLAYER_AP_HOOKSHOT;
+}
+
+s32 func_801240DC(Player* player) {
+    return Player_HoldsHookshot(player) && (player->heldActor == NULL);
+}
+
+s32 func_80124110(Player* player, s32 actionParam) {
+    s32 temp_v0 = actionParam - PLAYER_AP_UNK_2;
+
+    if (player->transformation != PLAYER_FORM_GORON) {
+        if (((actionParam - PLAYER_AP_UNK_2) > (PLAYER_AP_UNK_2 - PLAYER_AP_UNK_2)) &&
+            ((actionParam - PLAYER_AP_UNK_2) < (PLAYER_AP_SWORD_GREAT_FAIRY - PLAYER_AP_UNK_2))) {
+            return temp_v0;
+        }
+    }
+
+    return 0;
+}
+
+s32 func_80124148(Player* player) {
+    return func_80124110(player, player->itemActionParam);
+}
+
+s32 Player_ActionToMeleeWeapon(s32 actionParam) {
+    s32 weapon = actionParam - PLAYER_AP_UNK_2;
+
+    if ((weapon > (PLAYER_AP_UNK_2 - PLAYER_AP_UNK_2)) && (weapon <= (PLAYER_AP_UNK_8 - PLAYER_AP_UNK_2))) {
+        return weapon;
+    }
+    return 0;
+}
+
+s32 Player_GetMeleeWeaponHeld(Player* player) {
+    return Player_ActionToMeleeWeapon(player->itemActionParam);
+}
+
+s32 Player_HoldsTwoHandedWeapon(Player* player) {
+    // Relies on two-handed weapons to be contiguous
+    if ((player->itemActionParam >= PLAYER_AP_SWORD_GREAT_FAIRY) && (player->itemActionParam <= PLAYER_AP_STICK)) {
+        return true;
+    }
+
+    return false;
+}
+
+s32 Player_ActionToBottle(Player* player, s32 actionParam) {
+    s32 bottle = actionParam - PLAYER_AP_BOTTLE;
+
+    // Relies on bottle-related action params to be contiguous
+    if ((bottle >= (PLAYER_AP_BOTTLE - PLAYER_AP_BOTTLE)) && (bottle <= (PLAYER_AP_BOTTLE_FAIRY - PLAYER_AP_BOTTLE))) {
+        return bottle;
+    }
+
+    return -1;
+}
+
+s32 Player_GetBottleHeld(Player* Player) {
+    return Player_ActionToBottle(Player, Player->itemActionParam);
+}
+
+s32 Player_ActionToExplosive(Player* player, s32 actionParam) {
+    s32 explosive = actionParam - PLAYER_AP_BOMB;
+
+    // Relies on explosive-related action params to be contiguous
+    if ((explosive >= (PLAYER_AP_BOMB - PLAYER_AP_BOMB)) && (explosive <= (PLAYER_AP_BOMBCHU - PLAYER_AP_BOMB))) {
+        return explosive;
+    }
+
+    return -1;
+}
+
+s32 Player_GetExplosiveHeld(Player* player) {
+    return Player_ActionToExplosive(player, player->itemActionParam);
+}
+
+s32 func_80124278(Actor* actor, s32 arg1) {
+    s32 phi_v1 = 0;
+
+    // Fake match?
+    if ((arg1 == 1) || ((phi_v1 = arg1 - 3, (phi_v1 >= 0)) && (phi_v1 < 4))) {
```

Comment:
```suggestion
    //! FAKE:
    if ((arg1 == 1) || ((phi_v1 = arg1 - 3, (phi_v1 >= 0)) && (phi_v1 < 4))) {
```
We're starting to use this label to quickly find fake matches (or likely fake matches, you never know with devs)